updater.go

```diff
@@ -156,9 +158,14 @@ func RunUpdater(config *UpdaterConfig, datastore database.Datastore, st *stopper
 
 var errReceivedStopSignal = errors.New("stopped")
 
-func updateWhileRenewingLock(datastore database.Datastore, whoAmI string, isFirstUpdate bool, st *stopper.Stopper) (sleepDuration time.Duration, err error) {
+func updateWhileRenewingLock(datastore database.Datastore, whoAmI string, isFirstUpdate bool, st *stopper.Stopper) (err error) {
 	g, ctx := errgroup.WithContext(context.Background())
+	// done context is used when updater finishes and all other
+	// go rutines in group should finish too
+	doneCtx := context.Background()
+	doneCtx, done := context.WithCancel(doneCtx)
```

Comment:
Can we not wrap the existing context rather than creating a brand new unrelated one?

Reply:
@jzelinskie What do you mean by brand new context? I created a new context line above.

Reply:
Sorry for the confusion.

I'm suggesting this:
```go
var cancel context.CancelFunc
ctx, cancel = context.WithCancel(ctx) // <-- the same context from errgroup.WithContext()
```

Reply:
I am not sure if I understand your suggestion so I rather ask... I need 2 separate contexts to be able to distinguish between when an update is successfully done and when an update fails. You suggest merging those context together if I understand it right. How can we distinguish between those 2 cases then?